skyfield/positionlib.py

```diff
@@ -245,6 +272,8 @@ def altaz(self, temperature_C=None, pressure_mbar='standard'):
         """
         return _to_altaz(self.position.au, self.observer_data,
                          temperature_C, pressure_mbar)
+        
+    elements = elements
```

Comment:
This is an interesting pattern! I wonder if there will be any confusion, when users see tracebacks, about which class was involved in the method call? Once we merge I can do some testing on this to see the UX implications for exception names etc.

Reply:
Could ``Geocentric`` and``Barycentric`` positions could be subclasses of ``Geometric`` positions, since geocentric and barycentric positions are special cases of geometric positions? That would make this pattern unnecessary.

Reply:
That's an interesting question, and maybe I should review the classes sometime to see if I'm trying to fit too many different ideas into these classes. In general, the answer to your question is "no", because the vast majority of `Geocentric` and `Barycentric` positions are astrometric or apparent positions, not geometric (= non-apparent).